### PR TITLE
support reduction for 3d & 4d dims

### PR DESCRIPTION
**Pull request by asandhupatlaTT**
### Ticket
Link to Github Issue
https://github.com/tenstorrent/tt-metal/issues/16118

### Problem description
Provide context for the problem.
Current code doesn't support dim.size()>2

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.
Added support for higher dim.
Note: it only supports if input_tensor_rank == dim.size(). Other combinations are out of scope of this PR

### Checklist
- [x] Post commit CI passes : https://github.com/tenstorrent/tt-metal/actions/runs/12435618460
- [x] Blackhole Post commit (if applicable) : https://github.com/tenstorrent/tt-metal/actions/runs/12435616950
- [x] Model regression CI testing passes (if applicable) : https://github.com/tenstorrent/tt-metal/actions/runs/12435615910
- [x] Device performance regression CI testing passes (if applicable) : https://github.com/tenstorrent/tt-metal/actions/runs/12435614685
- [ ] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by asandhupatlaTT** [comment]
Note: it only supports if input_tensor_rank == dim.size(). Other combinations are out of scope of this PR

File: tests/ttnn/unit_tests/operations/test_reduction.py [hunk 1]
```diff
@@ -48,3 +48,25 @@ def test_var(device, batch_size, h, w, dim):
 
     output_tensor = ttnn.to_torch(output_tensor)
     assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.99)
+
+
+@pytest.mark.parametrize("batch_size", [1, 16, 1, 16])
+@pytest.mark.parametrize("c", [1, 8, 1, 16])
+@pytest.mark.parametrize("h", [32, 64, 41, 37])
+@pytest.mark.parametrize("w", [32, 64, 31, 63])
+@pytest.mark.parametrize("dim", [[0, 1, 2, 3]])
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/test_reduction.py:57]: dim should go through None, [], and [0,1,2,3]

File: tests/ttnn/unit_tests/operations/test_reduction.py [hunk 2]
```diff
@@ -48,3 +48,25 @@ def test_var(device, batch_size, h, w, dim):
 
     output_tensor = ttnn.to_torch(output_tensor)
     assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.99)
+
+
+@pytest.mark.parametrize("batch_size", [1, 16, 1, 16])
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/test_reduction.py:53]: Please don't have duplicates. Just `[1,16]`

File: tests/ttnn/unit_tests/operations/test_reduction.py [hunk 3]
```diff
@@ -48,3 +48,25 @@ def test_var(device, batch_size, h, w, dim):
...
+@pytest.mark.parametrize("c", [1, 8, 1, 16])
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/test_reduction.py:54]: Just `[1,8]`

**Review by bbradelTT — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/test_reduction.py (hunk 1 above)
Comment (bbradelTT) [reply tests/ttnn/unit_tests/operations/test_reduction.py:57]: Given the issue with [] please just leave as is.

File: tests/ttnn/unit_tests/operations/test_reduction.py (hunk 2 above)
Comment (bbradelTT) [reply tests/ttnn/unit_tests/operations/test_reduction.py:53]: Can be left for next PR.

File: tests/ttnn/unit_tests/operations/test_reduction.py (hunk 3 above)
Comment (bbradelTT) [reply tests/ttnn/unit_tests/operations/test_reduction.py:54]: Can be left for next PR.

File: tests/ttnn/unit_tests/operations/test_reduction.py (hunk 2 above)
Comment (asandhupatlaTT) [reply tests/ttnn/unit_tests/operations/test_reduction.py:53]: ohh yes. its typo. Will fix it

File: tests/ttnn/unit_tests/operations/test_reduction.py (hunk 1 above)
Comment (asandhupatlaTT) [reply tests/ttnn/unit_tests/operations/test_reduction.py:57]: adding only None for now. Will address [] in next PR

File: tests/ttnn/unit_tests/operations/test_reduction.py (hunk 3 above)
Comment (asandhupatlaTT) [reply tests/ttnn/unit_tests/operations/test_reduction.py:54]: fixed it next patch

**Pull request merged**